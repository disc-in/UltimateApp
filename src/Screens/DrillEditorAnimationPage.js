import React, { useState, useLayoutEffect } from 'react';
import { StyleSheet, View } from 'react-native';

import { DrillTypes } from '../Fixtures/config';
import theme from '../styles/theme.style';
import Drill from '../Components/animation/Drill';
import AnimationEditor from '../Components/editor/AnimationEditor';
import AnimationHistory from '../Components/editor/toolbar/AnimationHistory';
import HeaderButton from '../Components/shared/HeaderButton';

export const DrillEditorAnimationPage = ({ navigation, route }) => {
  const [currentAnimation, setCurrentAnimation] = useState(new Drill(route.params.animation));

  const onAnimationChange = (animation) => {
    setCurrentAnimation(animation);
    route.params.onAnimationChange(animation);
  };

  const handleSubmit = () => {
    navigation.navigate('DrillEditorPage');
  };

  useLayoutEffect(() => {
    navigation.setOptions({
      headerRight: () => <HeaderButton icon="check" onPress={handleSubmit} testID="validateButton" />,
    });
  });

  return (
    <View style={styles.drillEditorPage}>
      <View style={styles.centeredView}>
        <AnimationEditor onAnimationChange={onAnimationChange} animation={currentAnimation} uuid="unused_uuid" />
      </View>
      <View style={styles.toolBar}>
        <View style={styles.toolBarItem}>
          <AnimationHistory animation={currentAnimation} onAnimationHistoryChange={onAnimationChange} />
        </View>
      </View>
    </View>
  );
};

export default DrillEditorAnimationPage;

const styles = StyleSheet.create({
  drillEditorPage: {
    flex: 1,
  },
  centeredView: {
    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
    flex: 1,
  },
  toolBar: {
    height: '8%',
    minHeight: 40,
    width: '100%',
    backgroundColor: theme.COLOR_PRIMARY,
    position: 'absolute',
    bottom: 0,
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-evenly',
  },
  toolBarItem: {
    flex: 1,
    alignItems: 'center',
    paddingHorizontal: 10,
  },
  separator: {
    height: '70%',
    borderRightWidth: 0.5,
    borderRightColor: theme.COLOR_SECONDARY,
  },
});
