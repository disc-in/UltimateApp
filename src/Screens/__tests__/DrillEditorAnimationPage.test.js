import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

import DrillEditorAnimationPage from '../DrillEditorAnimationPage';

jest.mock('react-native-reanimated', () => require('react-native-reanimated/mock'));
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

describe('<DrillEditorAnimationPage />', () => {
  const route = {
    params: {
      onAnimationChange: jest.fn(),
      animation: undefined,
    },
  };

  it('renders correctly', async () => {
    const navigation = { navigate: jest.fn(), setOptions: jest.fn() };

    const { toJSON } = render(<DrillEditorAnimationPage route={route} navigation={navigation} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('goes back to the drill editor', async () => {
    const navigate = jest.fn();

    const Stack = createStackNavigator();
    const { getByTestId } = await waitFor(() =>
      render(
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen
              name="DrillEditorAnimationPage"
              component={DrillEditorAnimationPage}
              initialParams={{ route }}
              listeners={({ navigation }) => ({
                transitionStart: (e) => {
                  navigation.navigate = navigate;
                },
              })}
            />
          </Stack.Navigator>
        </NavigationContainer>,
      ),
    );

    fireEvent.press(getByTestId('validateButton'));

    expect(navigate).toBeCalledWith('DrillEditorPage');
  });
});
