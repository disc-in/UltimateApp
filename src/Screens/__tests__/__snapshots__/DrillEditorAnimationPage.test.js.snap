// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrillEditorAnimationPage /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Object {
              "height": 833.75,
              "width": 750,
            }
          }
        >
          <View
            style={
              Object {
                "height": 833.75,
                "width": 750,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#232b2b",
                    "borderWidth": 1,
                    "position": "absolute",
                  },
                  Object {
                    "height": 708.6875,
                    "left": 712.5,
                    "top": 41.6875,
                    "width": 3.375,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#232b2b",
                    "borderWidth": 1,
                    "position": "absolute",
                  },
                  Object {
                    "height": 708.6875,
                    "left": 37.5,
                    "top": 41.6875,
                    "width": 3.375,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#232b2b",
                    "borderWidth": 1,
                    "position": "absolute",
                  },
                  Object {
                    "height": 3.375,
                    "left": 37.5,
                    "top": 41.6875,
                    "width": 675,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#232b2b",
                    "borderWidth": 1,
                    "position": "absolute",
                  },
                  Object {
                    "height": 3.375,
                    "left": 37.5,
                    "top": 325.1625,
                    "width": 675,
                  },
                ]
              }
            />
            <Text />
          </View>
          <View
            height="100%"
            style={
              Object {
                "left": 0,
                "position": "absolute",
                "top": 0,
              }
            }
            width="100%"
          />
        </View>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingHorizontal": 10,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "marginHorizontal": 10,
                  "opacity": 1,
                }
              }
            >
              <Text />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "marginVertical": 10,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "gray",
                      "height": 5,
                      "position": "absolute",
                      "top": 10,
                    },
                    Object {
                      "width": 307.5,
                    },
                  ]
                }
              />
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "black",
                    "height": 5,
                    "position": "absolute",
                    "top": 10,
                    "width": 0,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "gray",
                      "borderRadius": 15,
                      "height": 15,
                      "position": "absolute",
                      "top": 5,
                      "width": 15,
                    },
                    Object {
                      "left": 0,
                    },
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "gray",
                      "borderRadius": 15,
                      "height": 15,
                      "position": "absolute",
                      "top": 5,
                      "width": 15,
                    },
                    Object {
                      "left": 307.5,
                    },
                  ]
                }
              />
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "black",
                    "borderRadius": 15,
                    "height": 15,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "top": 5,
                    "width": 15,
                  }
                }
              />
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "black",
                    "borderRadius": 15,
                    "height": 15,
                    "left": 307.5,
                    "opacity": 0,
                    "position": "absolute",
                    "top": 5,
                    "width": 15,
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "position": "absolute",
                      "top": 25,
                    },
                    Object {
                      "left": 3,
                    },
                  ]
                }
              >
                1
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "position": "absolute",
                      "top": 25,
                    },
                    Object {
                      "left": 310.5,
                    },
                  ]
                }
              >
                2
              </Text>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "height": "100%",
                    "left": -7.5,
                    "opacity": 1,
                    "position": "absolute",
                    "width": 30,
                  }
                }
              />
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "height": "100%",
                    "left": 300,
                    "opacity": 1,
                    "position": "absolute",
                    "width": 30,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "marginHorizontal": 10,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": "12%",
            "marginHorizontal": 10,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "flexDirection": "row",
              },
              Object {
                "alignItems": "center",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 30,
                "justifyContent": "space-around",
                "position": "absolute",
                "width": "100%",
                "zIndex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 80,
                  "width": "45%",
                }
              }
            >
              <RNCPicker
                items={
                  Array [
                    Object {
                      "label": "Rectangle",
                      "testID": undefined,
                      "textColor": undefined,
                      "value": "rectangle",
                    },
                    Object {
                      "label": "Endzone",
                      "testID": undefined,
                      "textColor": undefined,
                      "value": "endzone",
                    },
                    Object {
                      "label": "3/4 field",
                      "testID": undefined,
                      "textColor": undefined,
                      "value": "three quarters field",
                    },
                    Object {
                      "label": "Empty",
                      "testID": undefined,
                      "textColor": undefined,
                      "value": "empty",
                    },
                  ]
                }
                numberOfLines={1}
                onChange={[Function]}
                selectedIndex={1}
                style={
                  Array [
                    Object {
                      "height": 216,
                    },
                    Object {
                      "fontSize": 14,
                      "height": 80,
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                  "marginLeft": 20,
                  "width": "45%",
                }
              }
            >
              <View
                collapsable={false}
                onMoveShouldSetResponder={[Function]}
                onMoveShouldSetResponderCapture={[Function]}
                onResponderEnd={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderReject={[Function]}
                onResponderRelease={[Function]}
                onResponderStart={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onStartShouldSetResponderCapture={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexBasis": "25%",
                    "height": "100%",
                    "justifyContent": "center",
                    "minHeight": 0,
                    "minWidth": 0,
                    "padding": 5,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      },
                      Object {
                        "backgroundColor": "#2a9e91",
                      },
                      Object {
                        "borderRadius": 0,
                        "height": 0,
                        "width": 0,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#fff",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    1
                  </Text>
                </View>
              </View>
              <View
                collapsable={false}
                onMoveShouldSetResponder={[Function]}
                onMoveShouldSetResponderCapture={[Function]}
                onResponderEnd={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderReject={[Function]}
                onResponderRelease={[Function]}
                onResponderStart={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onStartShouldSetResponderCapture={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexBasis": "25%",
                    "height": "100%",
                    "justifyContent": "center",
                    "minHeight": 0,
                    "minWidth": 0,
                    "padding": 5,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      },
                      Object {
                        "backgroundColor": "#000000",
                      },
                      Object {
                        "borderRadius": 0,
                        "height": 0,
                        "width": 0,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#fff",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    1
                  </Text>
                </View>
              </View>
              <View
                collapsable={false}
                onMoveShouldSetResponder={[Function]}
                onMoveShouldSetResponderCapture={[Function]}
                onResponderEnd={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderReject={[Function]}
                onResponderRelease={[Function]}
                onResponderStart={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onStartShouldSetResponderCapture={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexBasis": "25%",
                    "height": "100%",
                    "justifyContent": "center",
                    "minHeight": 0,
                    "minWidth": 0,
                    "padding": 5,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "white",
                        "borderColor": "black",
                        "justifyContent": "center",
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      },
                      Object {
                        "borderRadius": 0,
                        "borderWidth": 0,
                        "height": 0,
                        "width": 0,
                      },
                    ]
                  }
                />
              </View>
              <View
                collapsable={false}
                onMoveShouldSetResponder={[Function]}
                onMoveShouldSetResponderCapture={[Function]}
                onResponderEnd={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderReject={[Function]}
                onResponderRelease={[Function]}
                onResponderStart={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onStartShouldSetResponderCapture={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexBasis": "25%",
                    "height": "100%",
                    "justifyContent": "center",
                    "minHeight": 0,
                    "minWidth": 0,
                    "padding": 5,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderBottomColor": "#feae19",
                        "borderColor": "transparent",
                        "borderStyle": "solid",
                        "justifyContent": "center",
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      },
                      Object {
                        "borderBottomWidth": 0,
                        "borderLeftWidth": 0,
                        "borderRightWidth": 0,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "height": 0,
                        "width": 0,
                      }
                    }
                  >
                    1
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#232b2b",
        "bottom": 0,
        "flexDirection": "row",
        "height": "8%",
        "justifyContent": "space-evenly",
        "minHeight": 40,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "paddingHorizontal": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexGrow": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID="undoButton"
        >
          <Text />
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexGrow": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID="redoButton"
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
</View>
`;
